### PR TITLE
Windowless sleep

### PR DESCRIPTION
**Pull request by futurepaul**
web-sys calls the "window" object for setTimeout, but that's not actually necessary (setTimeout is available in js environments without a window, like node). Also, it seems like the web-sys console_log is only used for tests, so with that we can move the web-sys dep entirely to dev-dependencies

this all seems to work fine in browser but I can't figure out how to run the tests on my mac

### CODE REVIEW

**Comment by benthecarman** [comment]
Is there a way we can do this without having to add javascript, or at least a library that does this for us?

**Comment by benthecarman** [comment]
This could be worth trying instead

https://docs.rs/async-std/latest/async_std/task/fn.sleep.html

**Comment by TonyGiorgio** [comment]
Agreed, no js. Make it into a rust crate and we'll add it :P 

> web-sys calls the "window" object for setTimeout,

Why is this a problem though?

**Comment by benthecarman** [comment]
For Javascript environments that aren't a browser

**Comment by futurepaul** [comment]
> This could be worth trying instead
> 
> https://docs.rs/async-std/latest/async_std/task/fn.sleep.html

does this work with wasm? tried looking at the impl but I couldn't quite follow it

**Comment by futurepaul** [comment]
> Agreed, no js. Make it into a rust crate and we'll add it :P
> 
> > web-sys calls the "window" object for setTimeout,
> 
> Why is this a problem though?

yeah this is helpful for WASI or node contexts, but also there's no "window" object in a web worker

**Comment by benthecarman** [comment]
> does this work with wasm? tried looking at the impl but I couldn't quite follow it

idk, was hoping you'd test :sweat_smile:

**Comment by futurepaul** [comment]
> idk, was hoping you'd test 😅

lol it's not if this tracking issue is real https://github.com/async-rs/async-std/issues/220

**Comment by futurepaul** [comment]
windowless_sleep = "0.1.1"

**Comment by TonyGiorgio** [comment]
Is this good to merge? It would be good to remove local storage.

**Comment by benthecarman** [comment]
Should be okay

File: mutiny-wasm/src/utils.rs [hunk 1]
```diff
@@ -27,16 +27,15 @@ pub fn now() -> Duration {
...
-    let p = js_sys::Promise::new(&mut cb);
-    wasm_bindgen_futures::JsFuture::from(p).await.unwrap();
+    #[cfg(target_arch = "wasm32")]
+    {
+        windowless_sleep::sleep(millis).await;
+    }
+    #[cfg(not(target_arch = "wasm32"))]
+    {
+        tokio::time::sleep(Duration::from_millis(millis.try_into().unwrap())).await;
+    }
```
Comment (TonyGiorgio) [inline mutiny-wasm/src/utils.rs:38]: This should only be targetting wasm32, so I'm not sure why tokio stuff is in here.

**Review by TonyGiorgio — CHANGES_REQUESTED** [review]

File: mutiny-wasm/src/utils.rs (hunk 1 above)
Comment (benthecarman) [reply mutiny-wasm/src/utils.rs:38]: we should be able to just get rid of this and re-export from mutiny core

**Review by TonyGiorgio — APPROVED** [review]

**Comment by TonyGiorgio** [comment]
Needs rebase.

**Pull request merged**